sys/riscv/include/vmparam.h

```diff
@@ -160,10 +160,32 @@
 #define	PHYS_IN_DMAP(pa)	((pa) >= DMAP_MIN_PHYSADDR && \
     (pa) < DMAP_MAX_PHYSADDR)
 /* True if va is in the dmap range */
+#ifdef __CHERI_PURE_CAPABILITY__
+#define	VIRT_IN_DMAP(va)						\
+	cheri_is_address_inbounds(dmap_capability, (va))
+#else
 #define	VIRT_IN_DMAP(va)	((va) >= DMAP_MIN_ADDRESS && \
     (va) < (dmap_max_addr))
+#endif
 
 #define	PMAP_HAS_DMAP	1
+#ifdef __CHERI_PURE_CAPABILITY__
+#define	PHYS_TO_DMAP(pa)						\
+({									\
+	KASSERT(PHYS_IN_DMAP(pa),					\
+	    ("%s: PA out of range, PA: 0x%lx", __func__,		\
+	    (vm_paddr_t)(pa)));						\
+	(vm_ptr_t)dmap_capability + ((pa) - dmap_phys_base);		\
+})
+
+#define	DMAP_TO_PHYS(va)						\
+({									\
+	KASSERT(VIRT_IN_DMAP(va),					\
+	    ("%s: VA out of range, VA: 0x%lx", __func__,		\
+	    (vm_offset_t)(va)));					\
+	dmap_phys_base + ((vm_offset_t)(va) - (vaddr_t)dmap_capability); \
+})
```

Comment:
Can we do something so DMAP_TO_PHYS share an implementation (e.g. introduce a `dmap_base` that's either `dmap_capability` or `DMAP_MIN_ADDRESS`.. or in fact you could just rename `dmap_capability` and have a fallback define, depends whether you think that's more confusing or not).

Reply:
Hmmm, part of this may have been that originally I had PHYS_TO_DMAP return a pointer rather than a `vm_pointer_t`.  Also, this is another reason I want to leave these headers out of this and instead pull it in as part of a RISC-V change to handle the direct map, etc.